### PR TITLE
Add an Append.Sequence instance for List

### LINKED ISSUE #2107: Cannot use += and ++= to Keys.developers
Type of Keys.developers is `SettingKey[List[Developer]]`. And there is no `Append.Sequence` instance for List.
This seems to needs to change the type to Seq or add the instance. The former is simple fix, but maybe breaks compatibility.

### PR DESCRIPTION
**Pull request by pdalpra**
Fixes #2107

### CODE REVIEW

**Comment by pdalpra** [comment]
Is there a place where I could add tests for this ?

**Comment by typesafe-tools** [comment]
Can one of the admins verify this patch?

**Comment by eed3si9n** [comment]
https://github.com/sbt/sbt/blob/0.13/main/settings/src/test/scala/UsageTest.scala looks like a place.

**Comment by jsuereth** [comment]
With tests, this would LGTM

**Comment by eed3si9n** [comment]
And notes.

**Comment by pdalpra** [comment]
Amended my commit to add notes and tests.

File: main/settings/src/test/scala/UsageTest.scala [hunk 1]
```diff
@@ -100,4 +102,14 @@ object Assign {
...
+
+  // Test that Append.Sequence instances for Seq/List work and don't mess up with each other
+  seqSetting := Seq("test1")
+  seqSetting ++= Seq("test2")
+  seqSetting ++= List("test3")
+  seqSetting += "test4"
+
+  seqSetting := List("test1")
+  seqSetting ++= List("test2")
+  seqSetting += "test4"
```
Comment (dwijnand) [inline main/settings/src/test/scala/UsageTest.scala]: I think these 3 should be `listSetting`.

Comment (pdalpra) [inline main/settings/src/test/scala/UsageTest.scala]: Woops.

Comment (pdalpra) [inline main/settings/src/test/scala/UsageTest.scala]: fixed

**Comment by dwijnand** [comment]
LGTM, unless we want to also test that the view bound variant is also working.

**Comment by jsuereth** [comment]
LGTM

**Pull request merged**